storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/JdbcStorageManager.java

```diff
@@ -81,6 +84,14 @@ public void addOrUpdate(Storable storable) throws StorageException {
     }
 
     @Override
+    public void update(Storable storable) {
+        if (queryExecutor.update(storable) == 0) {
```

Comment:
It could be,

1. The row existed during the query, but got deleted before it could be updated.
 In this case its a ConcurrentModification.
2. Wrong PK. Ideally this should not happen we expect only valid rows to be updated. i.e. the REST should have returned the row to the client which updated some non PK columns.

Its not really possible to differentiate b/w 1 and 2 from the storage manager and since we dont expect (2) in most of the cases, it should be ok to throw ConcurrentUpdateException. 

I will reword the exception message to include (2) as a possibility.